scripts/src/middleware.ts

```diff
@@ -3,21 +3,63 @@ import { LoggerInstance } from "winston";
 import * as bearerToken from "express-bearer-token";
 
 import { getLogger } from "./logging";
+import { generateId } from "./utils";
 
 let logger: LoggerInstance;
 export function init(app: express.Express) {
 	logger = getLogger();
 
+	app.use(requestLogger);
 	app.use(bearerToken());
 	app.use(logRequest);
 }
 
-export function logRequest(req: express.Request, res, next) {
-	logger.info(`start handling request: ${ req.method } ${ req.path } with ${ req.rawHeaders }`);
+declare module "express" {
+	interface Request {
+		readonly id: string;
+		readonly logger: LoggerInstance;
+	}
+}
+
+function requestLogger(req: express.Request, res, next) {
```

Comment:
maybe just add a comment here about how it works? i.e: if the last argument to the logger is an object, merge the reqId into it, otherwise add a new object with a single reqId key